### ISSUE TITLE
[Bug]: Some PHP errors fail silently and stop further tests execution

### ISSUE THREAD

**Issue by toonvandenbos** [pestphp/pest#1063]
### What Happened

While developing a new Laravel Package I stumbled across an unexpected "silent" failure in Pest's execution.

In one of my test files, I had a specific test that needed to instantiate a "Fixture" class as such:

```php

use Whitecube\NewPackage\Manager;
use Whitecube\NewPackage\Tests\Fixtures\FakeResolver;

it('can perform this example test', function () {
   $resolver = new FakeResolver();
   $service = new Manager($resolver);

   // ...
});
```

This `Whitecube\NewPackage\Tests\Fixtures\FakeResolver` class is implementing an interface my package requires, but at some point I forgot to implement one of its methods. I would expect Pest to tell me about it and show the exception as it does for any other problem encountered during the tests' execution, but instead it does nothing and stops any further testing execution.

To my knowledge, any other exception or error has an appropriate output. For instance, I tried to instantiate an undefined class inside said test, resulting in the expected exception output. It seems to me there only is a weird behavior when an interface's method is not implemented in a concrete class.

### How to Reproduce

1. Create a new "dummy package" directory: `mkdir foobar && cd foobar && mkdir src/ && touch src/FooBar.php`
2. Init composer:  <img width="746" alt="image" src="https://github.com/pestphp/pest/assets/5635557/d5d687e6-874e-406c-a4fb-fe8c77882d41">
3. Install and init `pestphp/pest` as described in the docs: `composer require pestphp/pest --dev --with-all-dependencies && ./vendor/bin/pest --init`
4. Create an interface inside the package's `src` directory:
    ```php
    namespace Whitecube\Foobar;

    interface SampleInterface
    {
        public function methodToImplement();
    }
    ```
5. Create a new fixture class implementing the freshly created interface inside the package's `tests/Fixtures` directory, but do not implement the interface's `methodToImplement` method:
    ```php
    namespace Whitecube\Foobar\Tests\Fixtures;

    use Whitecube\Foobar\SampleInterface;

    class FakeObject implements SampleInterface
    {
    }
    ```
6. Do not forget to add an autoloading rule in the `composer.json` file for the classes located inside `/tests`, then run `composer dumpautoload`: 
    ```json
    "autoload-dev": {
        "psr-4": {
            "Whitecube\\Foobar\\Tests\\Fixtures\\": "tests/Fixtures"
        }
    },
    ```
7. Write 3 dummy tests and let the middle one create a new instance of `Whitecube\Foobar\Tests\Fixtures\FakeObject`:
    ```php
    use Whitecube\Foobar\Tests\Fixtures\FakeObject;

    test('sample dummy test', function () {
        expect(true)->toBeTrue();
    });

    test('failing dummy test', function () {
        new FakeObject;
        expect(true)->toBeTrue();
    });

    test('another dummy test', function () {
        expect(true)->toBeTrue();
    });
    ```

Now, run the tests using `./vendor/bin/pest`. In my case, nothing is displayed in the output:
<img width="159" alt="image" src="https://github.com/pestphp/pest/assets/5635557/e50e8cd2-ec90-480a-afeb-140e6a791219">

In order to demonstrate that everything is working as expected, all 3 tests are executed when commenting the `new FakeObject;` line (or when resolving the issue by implementing the missing method inside `FakeObject`, of course):
<img width="270" alt="image" src="https://github.com/pestphp/pest/assets/5635557/7daeb20a-b519-4919-8be7-54f3beff838a">

For more convenience, I published a sample repository containing the case above.

### Sample Repository

https://github.com/whitecube/foobar-pestphp-issue-sample

### Pest Version

2.31.0

### PHP Version

8.1.14

### Operation System

macOS

**Comment by MSR2012** [comment]
I am getting the following error when tried to re-produce.

`PHP Fatal error:  Class App\Library\FakeObject contains 1 abstract method and must therefore be declared abstract or implement the remaining methods `

**Comment by nunomaduro** [comment]
This is fixed on Pest 2.33 - it will be released tomorrow morning.
